c/tskit/trees.c

```diff
@@ -452,6 +466,8 @@ tsk_treeseq_init(
     self->num_trees = (tsk_size_t) num_trees;
     self->discrete_genome = true;
     self->discrete_time = true;
+    self->min_time = -INFINITY;
```

Comment:
I think you need to start with min = positive inf and max = negative inf for this to work.